src/CodeGen_LLVM.cpp

```diff
@@ -3858,7 +3858,14 @@ void CodeGen_LLVM::codegen_asserts(const vector<const AssertStmt *> &asserts) {
 
     // Now switch on the bitmask to the correct failure
     Expr case_idx = cast<int32_t>(count_trailing_zeros(bitmask));
-    auto *switch_inst = builder->CreateSwitch(codegen(case_idx), no_errors_bb, asserts.size(), very_likely_branch);
+    llvm::SmallVector<uint32_t, 32> weights;
```

Comment:
Given that asserts.size() is at most 63, may as well use 64 as the template parameter for the size of the small vector.

Reply:
Done. (I hadn't realized we have this limit -- have we ever exceeded this limit in a real-world pipeline?)

Reply:
All the time, but there's an outer loop that processes blocks of 63 asserts. The general strategy is that we process blocks of asserts in batches of size 63. The conditions go into a bitmask (one bit is reserved to represent all asserts passing). We then count leading zeros to generate a switch index so that we can branch to the correct error handler for the first false condition.

Reply:
I should add: the nice thing about this is that by computing the conditions together without any control flow in the way, the checks get vectorized by llvm's SLP vectorizer.